include/fastdds/rtps/common/EntityId_t.hpp

```diff
@@ -345,7 +345,7 @@ struct hash<eprosima::fastrtps::rtps::EntityId_t>
         value[1] = k.value[1];
         value[0] = k.value[2];
 #endif // if FASTDDS_IS_BIG_ENDIAN_TARGET
-        return static_cast<std::size_t>(*reinterpret_cast<const uint32_t*>(&value));
+        return static_cast<std::size_t>(*reinterpret_cast<const unsigned char*>(&value));
```

Comment:
First, I think we had a very nasty bug here!

We wanted to use `value` instead of `&value`.

Second, I think the whole method could be changed to 

```
        return (static_cast<size_t>(k.value[0]) << 16) |
            (static_cast<size_t>(k.value[1]) << 8) |
            static_cast<size_t>(k.value[2]);
```

That code has the same functionality but is more portable and less error prone.

Reply:
I think you're completely right. I only looked at the type itself. Also, I think that the endianess still needs to be taken into account.

Reply:
Endianness is taken into account. What the original code tried to do was to return an unsigned integer with binary value `0x00AABBCC`, where `AA = k.value[0]`, `BB = k.value[1]` and `CC = k.value[2]`. This is what my proposed code will exactly do.